scripts/alerts/create_lists_to_email.ts

```diff
@@ -0,0 +1,96 @@
+//@ts-ignore
+import createsend from 'createsend-node';
+import fs from 'fs-extra';
+import path from 'path';
+
+const ALERT_LIST_CUSTOM_FIELD = "[alert_list_csv]";
+const ALERT_SIGNUPS_LIST_ID = "768d7917086d0c35bfa52b95af2a67e7";
+
+interface Subscriber {
+    EmailAddress: string,
+    Name: string,
+    Date: string,
+    State: 'Active' | 'Unconfirmed' | 'Unsubscribed' | 'Bounced' | 'Deleted',
+    CustomFields: Array<{ Key: string, Value: any }>,
+    ReadsEmailWith: string, // often empty
+}
+
+interface ActiveSubscriberResult {
+    Results: Array<Subscriber>,
+    ResultsOrderedBy: string,
+    OrderDirection: 'asc' | 'desc',
+    PageNumber: number,
+    PageSize: number,
+    RecordsOnThisPage: number,
+    TotalNumberOfRecords: number,
+    NumberOfPages: number,
+}
+
+async function getSubscribersForList(
+        pageToGet: number, api: any, list_id: string): Promise<ActiveSubscriberResult> {
+    return new Promise((resolve, reject) => {
+        api.lists.getActiveSubscribers(list_id, { page: pageToGet }, (err: any, result: ActiveSubscriberResult) => {
+            if (err) { reject(err); }
+            resolve(result);
+        });
+    });
+}
+
+/**
+ * Generates a file of fips and corresponding users to email with alerts
+ * for that fips.
+ *
+ * Takes in a file with {[fips: string]: {...data}}
+ *
+ * You can run via `yarn create-lists-to-email`
+ *
+ * Generates a file alerts/users-to-email.json containing an object of fips and
+ * users to send an email to. See the Alert interface below for schema.
+ */
+(async () => {
+
+    // Sample fips, will eventually be json loaded from a file
+    const locationsWithAlerts: { [fips: string]: { [key: string]: any } } = {
+        "06": { "overall": true },
+        "13127": { "overall": true }
+    }
+
+    // fips to object with a list of users and the previous meta data
+    const usersToEmail: { [fips: string]: Array<string> } = {};
+
+    // TODO: use oauth here and add secrets into github.
```

Comment:
Out of curiosity, why is oauth better than using an api key?  (I don't have much context, but an API key _seems_ simpler).

Reply:
oh hmm just general best practice to use oauth if it exsits